pkg/3scale/amp/component/system.go

```diff
@@ -76,19 +76,31 @@ const (
 )
 
 const (
-	SystemFileStoragePVCName = "system-storage"
+	SystemFileStoragePVCName      = "system-storage"
+	SystemFileStorageS3VolumeName = "file-storage-s3-config"
 )
 
 type System struct {
 	Options *SystemOptions
 }
 
+type SystemFileStorageType string
+
+const (
+	SystemFileStorageTypePVC = "pvc"
+	SystemFileStorageTypeS3  = "s3"
+)
+
 func NewSystem(options *SystemOptions) *System {
 	return &System{Options: options}
 }
 
 func (system *System) Objects() []common.KubernetesObject {
-	sharedStorage := system.SharedStorage()
+	objects := []common.KubernetesObject{}
+	if system.fileStorageIsPVC() {
```

Comment:
Now we only return the system's PVC shared storage as an object when the FileStorage PVC option is provided. This Objects method is used by the templates

Reply:
Why don't you add those comments in the code instead of PR's comments